spiffworkflow-backend/src/spiffworkflow_backend/models/spec_reference.py

```diff
@@ -0,0 +1,84 @@
+"""Message_model."""
+from dataclasses import dataclass
+
+from flask_bpmn.models.db import db
+from flask_bpmn.models.db import SpiffworkflowBaseDBModel
+from flask_marshmallow import Schema
+from marshmallow import INCLUDE
+
+
+@dataclass()
+class SpecReference:
+    """File Reference Information.
+
+    Includes items such as the process id and name for a BPMN,
+    or the Decision id and Decision name for a DMN file.  There may be more than
+    one reference that points to a particular file - if for instance, there are
+    three executable processes in a collaboration within a BPMN Diagram.
+    """
+
+    identifier: str  # The id of the process or decision.  "Process_1234"
+    display_name: str  # The name of the process or decision. "Invoice Submission"
+    process_model_id: str
+    type: str  # can be 'process' or 'decision'
+    file_name: str  # The name of the file where this process or decision is defined.
+    relative_path: str  # The path to the file.
+    has_lanes: bool  # If this is a process, whether it has lanes or not.
+    is_executable: bool  # Whether this process or decision is designated as executable.
+    is_primary: bool  # Whether this is the primary process of a process model
+    messages: dict  # Any messages defined in the same file where this process is defined.
+    correlations: dict  # Any correlations defined in the same file with this process.
+    start_messages: list  # The names of any messages that would start this process.
+
+
+class SpecReferenceCache(SpiffworkflowBaseDBModel):
```

Comment:
Add Model to the end of this since that's what we've been doing for all db model classes.